src/gc/impl/conservative/gc.d

```diff
@@ -2078,6 +2078,135 @@ struct Gcx
     ToScanStack!(ScanRange) toscanPrecise;
 
     /**
+     * Finds the offset between an index and the next set bit, up to our max.
+     *
+     * Returns: The offset between index and the next bit,
+     *          or -1 if no bits are set between index and max
+     */
+    int bitScanForwardOffset(size_t* bitArray, size_t index, size_t max) nothrow
```

Comment:
>They are recalculated for each call of bitScanForwardOffset, even though they are likely to be known in the previous call.

They should be changing each time bitScanForwardOffset is called. When it is called recursively, there should be a new index to start at. This should look better once it isn't a recursive function though.

>Could be constant, but still expensive. Aren't you just calculating position = index & 63?

I can't exactly say that I know that much about working directly with bits(never actually had to do it before this). I'll try that out, but I expect you are right.